### PR TITLE
Testing leaflet map

### PR DESCRIPTION
**Pull request by jedjas12**
PR for adding the Geospatial Maps FE stuff

### CODE REVIEW

File: salesforce/my-conservation-life/package.json [hunk 1]
```diff
@@ -11,7 +11,15 @@
   "dependencies": {},
   "devDependencies": {
     "@salesforce/eslint-config-lwc": "^0.3.0",
+<<<<<<< HEAD
```
Comment (exw4141) [inline salesforce/my-conservation-life/package.json:14]: Remove

File: salesforce/my-conservation-life/force-app/main/default/lwc/geoQuery/geoQuery.js [hunk 2]
```diff
@@ -1,26 +1,27 @@
+/* eslint-disable vars-on-top */
 /* eslint-disable no-console */
 import { LightningElement } from 'lwc';
 
+import leafletDraw from '@salesforce/resourceUrl/leafletDraw';
+
 import utils from 'c/utils';
 
 const DISTANCE_URL = utils.URL + 'assets/geometrySearch/distance';
+const POLY_URL = utils.URL + '/assets/geometrySearch/polygon';
```
Comment (APalumbo97) [inline salesforce/my-conservation-life/force-app/main/default/lwc/geoQuery/geoQuery.js:10]: Either remove the first '/' or add it to the above line for consistency

File: salesforce/my-conservation-life/force-app/main/default/lwc/geoQuery/geoQuery.js [hunk 3]
```diff
@@ -36,7 +37,8 @@ export default class GeoQuery extends LightningElement {
      * @param {CustomEvent} event
      * @param {Map} event.details - Leaflet Map of the child component
      */
-    onMapInitialized(event) {
+
```
Comment (APalumbo97) [inline salesforce/my-conservation-life/force-app/main/default/lwc/geoQuery/geoQuery.js:40]: Remove newline between comment and function

File: salesforce/my-conservation-life/force-app/main/default/lwc/map/map.js [hunk 4]
```diff
@@ -1,16 +1,16 @@
...
+import leafletDraw from '@salesforce/resourceUrl/leafletDraw';
 
 /* L is the Leaflet object constructed by the leaflet.js script */
 /*global L*/
 
 export default class Map extends LightningElement {
 
     map; // L.map, a leaflet map. constructed in initializeleaflet
-
+
```
Comment (APalumbo97) [inline salesforce/my-conservation-life/force-app/main/default/lwc/map/map.js:13]: Remove whitespace

File: salesforce/my-conservation-life/force-app/main/default/staticresources/leafletdraw.resource-meta.xml [hunk 5]
```diff
@@ -0,0 +1,5 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<StaticResource xmlns="http://soap.sforce.com/2006/04/metadata">
+    <cacheControl>Public</cacheControl>
+    <contentType>application/zip</contentType>
+</StaticResource>
```
Comment (APalumbo97) [inline salesforce/my-conservation-life/force-app/main/default/staticresources/leafletdraw.resource-meta.xml:5]: Add EOF newline

File: salesforce/my-conservation-life/package.json (hunk 1 above)
Comment (APalumbo97) [reply salesforce/my-conservation-life/package.json:14]: +1

Comment (jedjas12) [reply salesforce/my-conservation-life/package.json]: Resolved

File: salesforce/my-conservation-life/force-app/main/default/lwc/geoQuery/geoQuery.js (hunk 2 above)
Comment (jedjas12) [reply salesforce/my-conservation-life/force-app/main/default/lwc/geoQuery/geoQuery.js]: Resolved

File: salesforce/my-conservation-life/force-app/main/default/staticresources/leafletdraw.resource-meta.xml (hunk 5 above)
Comment (jedjas12) [reply salesforce/my-conservation-life/force-app/main/default/staticresources/leafletdraw.resource-meta.xml:5]: Not my file - part of the packages.

File: salesforce/my-conservation-life/force-app/main/default/lwc/map/map.js (hunk 4 above)
Comment (jedjas12) [reply salesforce/my-conservation-life/force-app/main/default/lwc/map/map.js:13]: Resolved

File: salesforce/my-conservation-life/force-app/main/default/lwc/geoQuery/geoQuery.js [hunk 6]
```diff
@@ -36,7 +37,7 @@ export default class GeoQuery extends LightningElement {
      * @param {CustomEvent} event
      * @param {Map} event.details - Leaflet Map of the child component
      */
-    onMapInitialized(event) {
+     onMapInitialized(event) {
```
Comment (APalumbo97) [inline salesforce/my-conservation-life/force-app/main/default/lwc/geoQuery/geoQuery.js:40]: Whitespace

File: salesforce/my-conservation-life/force-app/main/default/lwc/map/map.js (hunk 4 above)
Comment (APalumbo97) [reply salesforce/my-conservation-life/force-app/main/default/lwc/map/map.js:12]: @jedjas12 It is still showing up

File: salesforce/my-conservation-life/force-app/main/default/lwc/map/map.js [hunk 7]
```diff
@@ -1,26 +1,27 @@
...
      */
-    
     connectedCallback() {
         Promise.all([
             loadScript(this, leaflet + '/leaflet.js'),
-            loadStyle(this, leaflet + '/leaflet.css')
+            loadStyle(this, leaflet + '/leaflet.css'),
+            loadScript(this, leafletDraw + '/dist/leaflet.draw.js'),
+            loadStyle(this, leafletDraw + '/dist/leaflet.draw.css')
+
```
Comment (APalumbo97) [inline salesforce/my-conservation-life/force-app/main/default/lwc/map/map.js:24]: Whitespace

File: db/api/v1/routes/validate.js [hunk 8]
```diff
@@ -345,6 +345,36 @@ const parseLongitude = (lonStr) => {
         ParseResult.failure('Longitudes must be numbers between -180 and 180');
 };
 
+/**
+ * Parse a latitude value from a string
+ * 
+ * @param {string} latStr - A string that is parsable into a floating point number
+ * @returns {ParseResult} - Parse success with a float value that is a latitude, or a failure message
+ */
+const parseLatitude = (latStr) => {
```
Comment (tilley14) [inline db/api/v1/routes/validate.js:354]: Duplicate code

File: db/api/v1/routes/validate.js [hunk 9]
```diff
@@ -345,6 +345,36 @@ const parseLongitude = (lonStr) => {
...
+        ParseResult.failure('Latitudes must be numbers between -90 and 90');
+};
+
+/**
+ * Parse a longitude value from a string
+ * 
+ * @param {string} lonStr - A string that is parsable into a floating point number
+ * @returns {ParseResult} - Parse success with a float value that is a longitude, or a failure message
+ */
+const parseLongitude = (lonStr) => {
```
Comment (tilley14) [inline db/api/v1/routes/validate.js:369]: Duplicate code

File: salesforce/my-conservation-life/force-app/main/default/lwc/geoQuery/geoQuery.js [hunk 10]
```diff
@@ -1,26 +1,27 @@
...
-
-    
+  
     /**
      * Starts the download for asset details and bounding box early.
      */
     connectedCallback() {
-        // this.assetsPromise = assets.find({ projectId: parseInt(this.projectId, 10) });
-        // this.assetsBboxPromise = bboxAssets.get(parseInt(this.projectId, 10));
+        this.assetsPromise = assets.find({ projectId: parseInt(this.projectId, 10) });
```
Comment (tilley14) [inline salesforce/my-conservation-life/force-app/main/default/lwc/geoQuery/geoQuery.js:23]: Not sure if we want to do these two methods, probably remove both lines...

File: salesforce/my-conservation-life/force-app/main/default/lwc/geoQuery/geoQuery.js [hunk 11]
```diff
@@ -50,96 +51,142 @@ export default class GeoQuery extends LightningElement {
...
+
+                    // Initialize the array if need be
+                    if (this.myAssets === undefined) {
+                        this.myAssets = [];
+                    }
+
+                    // Iterate over the assets and add them to the map
+                    let i;
+                    for (i = 0; i < response.length; i++) {
+                        const a = response[i];
```
Comment (tilley14) [inline salesforce/my-conservation-life/force-app/main/default/lwc/geoQuery/geoQuery.js:113]: could use a more descriptive variable name like asset

File: salesforce/my-conservation-life/force-app/main/default/lwc/geoQuery/geoQuery.js [hunk 12]
```diff
@@ -50,96 +51,142 @@ export default class GeoQuery extends LightningElement {
...
+
+                        // Create the map marker, asset type determine the icon
+                        let m;
```
Comment (tilley14) [inline salesforce/my-conservation-life/force-app/main/default/lwc/geoQuery/geoQuery.js:116]: same here with marker

File: salesforce/my-conservation-life/force-app/main/default/lwc/geoQuery/geoQuery.js [hunk 13]
```diff
@@ -50,96 +51,142 @@ export default class GeoQuery extends LightningElement {
...
+
+                    // Initialize the array if need be
+                    if (this.myAssets === undefined) {
+                        this.myAssets = [];
+                    }
+
+                    // Iterate over the assets and add them to the map
+                    let i;
+                    for (i = 0; i < response.length; i++) {
+                        const a = response[i];
```
Comment (tilley14) [inline salesforce/my-conservation-life/force-app/main/default/lwc/geoQuery/geoQuery.js:170]: Same as above, asset and marker

**Pull request merged**